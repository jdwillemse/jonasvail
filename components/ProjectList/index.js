import React, { useState, useEffect, useRef, useCallback } from 'react'
import { get } from 'lodash'
import { useRouter } from 'next/router'
import { debounce } from 'lodash'

import css from './styles.module.scss'
import ProjectItem from '../ProjectItem'

const ProjectList = ({ allProjects = [] }) => {
  const [list, setList] = useState(allProjects)
  const router = useRouter()
  const gridRef = useRef()
  const allItems = useRef()
  const { client } = router.query
  // loop over grid items and calculate size for each to conclude container height
  const debounced = useRef(
    debounce(() => {
      const winWidth = window.innerWidth
      const columnCount = winWidth >= 768 ? 3 : 2
      const columnHeight = Array.from(gridRef.current.children).reduce(
        (sum, element) => sum + element.getBoundingClientRect().height,
        0
      )

      gridRef.current.style.height = `${
        columnHeight / columnCount + winWidth * 0.4
      }px`
    }, 200)
  )

  // pattern from https://stackoverflow.com/questions/54666401/how-to-use-throttle-or-debounce-with-react-hook
  // eslint-disable-next-line react-hooks/exhaustive-deps
  const resizeAllGridItems = useCallback(debounced.current, [gridRef])

  // filter list of grid items based on client
  useEffect(() => {
    if (!client) {
      setList(allProjects)
    } else {
      const filter = client?.toLowerCase()
      const filteredList = allProjects.filter(
        (item) => get(item, 'node.client.name', '').toLowerCase() === filter
      )
      setList(filteredList)
    }
  }, [allProjects, client])

  // cache grid item list and add window resize listener
  useEffect(() => {
    window.addEventListener('resize', resizeAllGridItems)
    return () => {
      window.removeEventListener('resize', resizeAllGridItems)
    }
  }, [resizeAllGridItems])

  // update masonry layout if list updates
  useEffect(() => {
    if (!allItems.current) {
      allItems.current = gridRef.current.children
    }
    setTimeout(() => {
      resizeAllGridItems()
    }, 1000)
  }, [list, resizeAllGridItems])

  return (
    <section className={css.wrap}>
      <div className={css.masonry} ref={gridRef}>
        {list.map((project) => (
          <div className={css.item} key={project.node._meta.id}>
            <ProjectItem {...project} />
          </div>
        ))}
      </div>
    </section>
  )
}

export default ProjectList
