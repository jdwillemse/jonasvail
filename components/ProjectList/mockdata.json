[
  {
    "node": {
      "_meta": {
        "id": "X5Be0hIAACEAfgX0",
        "uid": "twitch-creators-campaign",
        "type": "project"
      },
      "title": [
        { "type": "heading1", "text": "Twitch Creators Campaign", "spans": [] }
      ],
      "listImage": {
        "dimensions": { "width": 1200, "height": 675 },
        "alt": null,
        "copyright": null,
        "url": "https://images.prismic.io/jonasvail/c95286cf-7de8-44c0-90f0-9122f77fe5db_Twitch+logo.jpg?auto=compress,format"
      },
      "date": "2020-09-28",
      "client": { "name": "Twitch" },
      "previewImage": "data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAACAAQDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAf/xAAfEAAABQQDAAAAAAAAAAAAAAAAAQIDBAUGERITITH/xAAVAQEBAAAAAAAAAAAAAAAAAAADBP/EABgRAAIDAAAAAAAAAAAAAAAAAAACAzEy/9oADAMBAAIRAxEAPwCHX9UJjrVr8suQvWiR0p2cUeCI14Iu/AABQ9iSaP/Z"
    }
  },
  {
    "node": {
      "_meta": {
        "id": "XpCc9RAAAO7Gzs__",
        "uid": "adidas-futurecraft-loop",
        "type": "project"
      },
      "title": [
        { "type": "heading1", "text": "[FUTURECRAFT] Loop", "spans": [] }
      ],
      "listImage": {
        "dimensions": { "width": 7000, "height": 4950 },
        "alt": null,
        "copyright": null,
        "url": "https://images.prismic.io/jonasvail/070c2ce5-ca64-4565-bc5e-1ef307689489_551468.jpg?auto=compress,format"
      },
      "date": "2019-04-10",
      "client": { "name": "adidas" },
      "previewImage": "data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAADAAQDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAj/xAAdEAACAgMAAwAAAAAAAAAAAAABAgMRAAQFEzFB/8QAFAEBAAAAAAAAAAAAAAAAAAAAAf/EABURAQEAAAAAAAAAAAAAAAAAAAEA/9oADAMBAAIRAxEAPwCjOHweZpQbCQaoKybEkp8rtJTMbNFiaF+lFAfAMYxirF//2Q=="
    }
  },
  {
    "node": {
      "_meta": {
        "id": "XpBkZRAAAO7Fzc8p",
        "uid": "ps4-this-could-be-you",
        "type": "project"
      },
      "title": [
        { "type": "heading1", "text": "This Could Be You", "spans": [] }
      ],
      "listImage": {
        "dimensions": { "width": 1280, "height": 720 },
        "alt": null,
        "copyright": null,
        "url": "https://images.prismic.io/jonasvail/2dc740e4-ee89-492a-be4a-9c10b727f1d2_This+could+be+you+Fortnite+skydiver.jpg?auto=compress,format"
      },
      "date": "2018-11-15",
      "client": { "name": "PlayStation" },
      "previewImage": "data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAADAAQDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAf/xAAfEAACAQQCAwAAAAAAAAAAAAABAgQAAwURBxIhIpH/xAAVAQEBAAAAAAAAAAAAAAAAAAADBP/EABoRAAICAwAAAAAAAAAAAAAAAAABAgMSITL/2gAMAwEAAhEDEQA/AKDxS97J4GdJyEqXJvnIyEDXZFxuqq3VVXZ9VAA8DQ+0pSo7e2BCTxWz/9k="
    }
  },
  {
    "node": {
      "_meta": {
        "id": "XpCb4RAAACzKzssV",
        "uid": "detroit-how-far-would-you-go-to-be-free",
        "type": "project"
      },
      "title": [
        {
          "type": "heading1",
          "text": "How far would you go to be free?",
          "spans": []
        }
      ],
      "listImage": {
        "dimensions": { "width": 1200, "height": 675 },
        "alt": null,
        "copyright": null,
        "url": "https://images.prismic.io/jonasvail/7f0c56d9-2181-4098-b505-73a28d836144_detroit-become-human.jpg?auto=compress,format"
      },
      "date": "2018-04-13",
      "client": { "name": "PlayStation" },
      "previewImage": "data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAACAAQDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAf/xAAeEAABBAIDAQAAAAAAAAAAAAAEAAECAwUGERIiMf/EABUBAQEAAAAAAAAAAAAAAAAAAAIE/8QAFxEBAAMAAAAAAAAAAAAAAAAAAAECMf/aAAwDAQACEQMRAD8Ahpe7bXjzixwNmzgo8b7OtVJ9sIt6f4zS4REV9tkX/9k="
    }
  },
  {
    "node": {
      "_meta": {
        "id": "XpBfkhAAAO7FzbkU",
        "uid": "gt-sport-where-the-world-comes-to-race",
        "type": "project"
      },
      "title": [
        {
          "type": "heading1",
          "text": "Where the World Comes to Race",
          "spans": []
        }
      ],
      "listImage": {
        "dimensions": { "width": 780, "height": 780 },
        "alt": null,
        "copyright": null,
        "url": "https://images.prismic.io/jonasvail/78bf9234-89d0-4bd0-86e4-997e0816fc53_SIEE_GT_SPORT_WTWCTR_2017_MCLAREN_VSN_30_w4KTAG+%281%29.jpeg?auto=compress,format"
      },
      "date": "2017-11-23",
      "client": { "name": "PlayStation" },
      "previewImage": "data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAAEAAQDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAf/xAAfEAABAwMFAAAAAAAAAAAAAAABAAIFAwQRBgcSIUH/xAAVAQEBAAAAAAAAAAAAAAAAAAADBf/EABcRAAMBAAAAAAAAAAAAAAAAAAABAhH/2gAMAwEAAhEDEQA/AKftxGQ91pGzNaBh3GmXUml1oxx4tOB2c+IiKbLeCn//2Q=="
    }
  }
]
