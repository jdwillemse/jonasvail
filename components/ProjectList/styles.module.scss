@import '../../styles/vars.scss';

.wrap {
  margin: 1rem 0;

  @media (min-width: $break-md) {
    margin: 3rem 0;
  }
}

// masonry grid from https://tobiasahlin.com/blog/masonry-with-css/
.masonry {
  display: flex;
  flex-flow: column wrap;
  align-content: space-between;
  /* Your container needs a fixed height, and it 
   * needs to be taller than your tallest column. */
  height: 1000vw;

  /* Force new columns */
  &::before,
  &::after {
    content: '';
    flex-basis: 100%;
    width: 0;
    order: 2;
  }
}

.item {
  width: calc(50% - 1rem);

  /* Re-order items into 2 rows */
  &:nth-child(2n + 1) {
    order: 1;
  }
  &:nth-child(2n + 2) {
    order: 2;
  }

  @media (min-width: 768px) {
    width: calc(33.33334% - 1rem);

    /* Re-order items into 3 rows */
    &:nth-child(3n + 1) {
      order: 1;
    }
    &:nth-child(3n + 2) {
      order: 2;
    }
    &:nth-child(3n) {
      order: 3;
    }
  }
}
