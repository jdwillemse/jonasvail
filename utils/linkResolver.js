export default function linkResolver ({ type, uid } = {}) {
  switch (type) {

    case 'project':
      return `/${uid}`

    default:
      return `/${uid}`
  }
}
