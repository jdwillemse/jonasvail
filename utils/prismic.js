import PropTypes from 'prop-types'
import React from 'react'
import { PrismicRichText } from '@prismicio/react'
import * as prismic from '@prismicio/client'

import linkResolver from './linkResolver'

// prop-types
export const structuredTextPropType = PropTypes.arrayOf(
  PropTypes.shape({
    type: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    spans: PropTypes.array.isRequired,
  })
)

export const imagePropType = PropTypes.shape({
  dimensions: PropTypes.shape({
    width: PropTypes.number.isRequired,
    height: PropTypes.number.isRequired,
  }).isRequired,
  alt: PropTypes.oneOfType([PropTypes.string]),
  copyright: PropTypes.oneOfType([PropTypes.string]),
  url: PropTypes.string.isRequired,
})

export const imageSharpPropType = PropTypes.shape({
  childImageSharp: {
    fluid: {
      aspectRatio: 1.565217391304348,
      sizes: PropTypes.string,
      src: PropTypes.string,
      srcSet: PropTypes.string,
      srcSetWebp: PropTypes.string,
      srcWebp: PropTypes.string,
    },
  },
})

//  rendering
export const asText = (data) => {
  if (!data) {
    return null
  }
  return typeof data === 'string' ? data : prismic.asText(data)
}

export const renderer = (data) =>
  data ? <PrismicRichText field={data} linkResolver={linkResolver} /> : null

const options = {
  year: 'numeric',
  month: 'short',
  day: 'numeric',
}
