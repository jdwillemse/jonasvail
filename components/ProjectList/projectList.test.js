import { render } from '@testing-library/react'

import ProjectList from '.'
import data from './mockdata.json'
import { mockNextUseRouter } from '../../utils/testUtils'

describe('ProjectList', () => {
  beforeEach(() => {
    // Mocks Next.js route
    mockNextUseRouter({
      route: '/',
      pathname: '/',
      query: {},
      asPath: '',
    })
  })

  it('renders a list of projects', () => {
    const { container } = render(<ProjectList allProjects={data} />)
    expect(container).toMatchSnapshot()
  })

  it('filters the list based on the route', () => {
    const clientName = 'playstation'
    mockNextUseRouter({
      route: '/',
      pathname: '/',
      query: { client: clientName },
      asPath: '',
    })
    const { container } = render(<ProjectList allProjects={data} />)
    container.querySelectorAll('.client').forEach((element) => {
      expect(element.textContent).toMatchSnapshot(clientName)
    })
  })

  it('does not throw errors when content is not passed', () => {
    const { container } = render(<ProjectList />)
    expect(
      container.querySelector('[data-test-id="masonry-wrap"]').childNodes.length
    ).toBe(0)
  })
})
