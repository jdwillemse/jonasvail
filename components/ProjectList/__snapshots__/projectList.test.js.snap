// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList filters the list based on the route: playstation 1`] = `"Twitch"`;

exports[`ProjectList filters the list based on the route: playstation 2`] = `"adidas"`;

exports[`ProjectList filters the list based on the route: playstation 3`] = `"PlayStation"`;

exports[`ProjectList filters the list based on the route: playstation 4`] = `"PlayStation"`;

exports[`ProjectList filters the list based on the route: playstation 5`] = `"PlayStation"`;

exports[`ProjectList renders a list of projects 1`] = `
<div>
  <section
    class="wrap"
  >
    <div
      class="masonry"
      data-test-id="masonry-wrap"
    >
      <div
        class="item"
      >
        <article
          class="wrap"
        >
          <a
            class="link"
            href="/twitch-creators-campaign"
          >
            <div
              class="imageWrap"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAACAAQDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAf/xAAfEAAABQQDAAAAAAAAAAAAAAAAAQIDBAUGERITITH/xAAVAQEBAAAAAAAAAAAAAAAAAAADBP/EABgRAAIDAAAAAAAAAAAAAAAAAAACAzEy/9oADAMBAAIRAxEAPwCHX9UJjrVr8suQvWiR0p2cUeCI14Iu/AABQ9iSaP/Z); background-position: 0% 0%;"
                />
                <noscript />
              </span>
            </div>
            <h1
              class="title"
            >
              Twitch Creators Campaign
            </h1>
            <div
              class="client"
            >
              Twitch
            </div>
          </a>
        </article>
      </div>
      <div
        class="item"
      >
        <article
          class="wrap"
        >
          <a
            class="link"
            href="/adidas-futurecraft-loop"
          >
            <div
              class="imageWrap"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 70.71428571428572% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAADAAQDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAj/xAAdEAACAgMAAwAAAAAAAAAAAAABAgMRAAQFEzFB/8QAFAEBAAAAAAAAAAAAAAAAAAAAAf/EABURAQEAAAAAAAAAAAAAAAAAAAEA/9oADAMBAAIRAxEAPwCjOHweZpQbCQaoKybEkp8rtJTMbNFiaF+lFAfAMYxirF//2Q==); background-position: 0% 0%;"
                />
                <noscript />
              </span>
            </div>
            <h1
              class="title"
            >
              [FUTURECRAFT] Loop
            </h1>
            <div
              class="client"
            >
              adidas
            </div>
          </a>
        </article>
      </div>
      <div
        class="item"
      >
        <article
          class="wrap"
        >
          <a
            class="link"
            href="/ps4-this-could-be-you"
          >
            <div
              class="imageWrap"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAADAAQDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAf/xAAfEAACAQQCAwAAAAAAAAAAAAABAgQAAwURBxIhIpH/xAAVAQEBAAAAAAAAAAAAAAAAAAADBP/EABoRAAICAwAAAAAAAAAAAAAAAAABAgMSITL/2gAMAwEAAhEDEQA/AKDxS97J4GdJyEqXJvnIyEDXZFxuqq3VVXZ9VAA8DQ+0pSo7e2BCTxWz/9k=); background-position: 0% 0%;"
                />
                <noscript />
              </span>
            </div>
            <h1
              class="title"
            >
              This Could Be You
            </h1>
            <div
              class="client"
            >
              PlayStation
            </div>
          </a>
        </article>
      </div>
      <div
        class="item"
      >
        <article
          class="wrap"
        >
          <a
            class="link"
            href="/detroit-how-far-would-you-go-to-be-free"
          >
            <div
              class="imageWrap"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 56.25% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAACAAQDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAf/xAAeEAABBAIDAQAAAAAAAAAAAAAEAAECAwUGERIiMf/EABUBAQEAAAAAAAAAAAAAAAAAAAIE/8QAFxEBAAMAAAAAAAAAAAAAAAAAAAECMf/aAAwDAQACEQMRAD8Ahpe7bXjzixwNmzgo8b7OtVJ9sIt6f4zS4REV9tkX/9k=); background-position: 0% 0%;"
                />
                <noscript />
              </span>
            </div>
            <h1
              class="title"
            >
              How far would you go to be free?
            </h1>
            <div
              class="client"
            >
              PlayStation
            </div>
          </a>
        </article>
      </div>
      <div
        class="item"
      >
        <article
          class="wrap"
        >
          <a
            class="link"
            href="/gt-sport-where-the-world-comes-to-race"
          >
            <div
              class="imageWrap"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                />
                <img
                  alt=""
                  data-nimg="responsive"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAAEAAQDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAf/xAAfEAABAwMFAAAAAAAAAAAAAAABAAIFAwQRBgcSIUH/xAAVAQEBAAAAAAAAAAAAAAAAAAADBf/EABcRAAMBAAAAAAAAAAAAAAAAAAABAhH/2gAMAwEAAhEDEQA/AKftxGQ91pGzNaBh3GmXUml1oxx4tOB2c+IiKbLeCn//2Q==); background-position: 0% 0%;"
                />
                <noscript />
              </span>
            </div>
            <h1
              class="title"
            >
              Where the World Comes to Race
            </h1>
            <div
              class="client"
            >
              PlayStation
            </div>
          </a>
        </article>
      </div>
    </div>
  </section>
</div>
`;
